Automattic/wp-calypso
# Podcasting: Add category selector

This PR adds a category selector to the podcast settings page using the `TermTreeSelector` component.  Follow-up to #23534 and part of p3Ex-32D-p2.

<img width="1047" alt="podcast-category" src="https://user-images.githubusercontent.com/942359/37932662-1909feac-3117-11e8-985b-d8c024c24367.png">

**To test:**
Go to `/settings/podcasting/:site_id` with the `manage/site-settings/podcasting` flag enabled (it will be enabled in development on `master`).

- Verify that a site with podcasting enabled loads with the correct category selected.
- Verify that changing the selected category marks the settings form as "dirty" (you get a prompt to confirm leaving the page).
- Verify that the selected category is saved as expected.
- Verify that if an invalid category slug is saved in the `podcasting_archive` option (the category was renamed, for example), then no category is selected.
- Verify that clicking "Add a new category" allows you to add a new category that is selected when the dialog closes.
